### PR TITLE
Add new graph ops endpoints

### PR DESCRIPTION
**Pull request by soerenreichardt**

### CODE REVIEW

**Comment by soerenreichardt** [comment]
[Update the python client](https://trello.com/c/YMHV3KTF/6439-update-the-python-client)

**Review by adamnsch — CHANGES_REQUESTED** [review]
Nice work :) Just have some minor comments

File: changelog/1.3.0.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 ## New features
 
 * Add MLP training method `addMLP` to link prediction and node classification pipelines.
+* Add new graph operations endpoints in gds >2.2.0
```
Comment (adamnsch) [inline changelog/1.3.0.md:10]:
```suggestion
* Add support for new graph catalog endpoints in GDS >= 2.2.0.
```

File: graphdatascience/graph/graph_ops_runner.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
+from typing import Any, Callable, Dict, List, Union
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/graph/graph_ops_runner.py:1]: I agree, but I struggle a bit with finding a good name. Maybe `graph_entity_ops_runner`?

File: graphdatascience/tests/unit/test_graph_ops.py [hunk 3]
```diff
@@ -229,6 +306,49 @@ def test_graph_streamRelationshipProperties(runner: CollectingQueryRunner, gds:
...
+                "sourceNodeId": 0,
+                "targetNodeId": 1,
+                "relationshipType": "REL",
+                "relationshipProperty": "dummyProp",
+                "propertyValue": 2,
+            }
+        ]
+    )
+
+    runner.set__mock_result(result_df)
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/tests/unit/test_graph_ops.py:325]: I copied this from the test above, but I can remove it in both cases

File: graphdatascience/tests/integration/test_graph_ops.py [hunk 4]
```diff
@@ -167,20 +170,30 @@ def test_graph_streamNodeProperty_with_arrow_no_db(gds: GraphDataScience) -> Non
...
+        else:
+            gds.graph.nodeProperty.stream(G, "x", concurrency=2)
 
 
 def test_graph_streamNodeProperty_without_arrow(gds_without_arrow: GraphDataScience) -> None:
     G, _ = gds_without_arrow.graph.project(GRAPH_NAME, {"Node": {"properties": "x"}}, "*")
 
-    result = gds_without_arrow.graph.streamNodeProperty(G, "x", concurrency=2)
+    if gds_without_arrow._server_version < ServerVersion(2, 2, 0):
+        result = gds_without_arrow.graph.streamNodeProperty(G, "x", concurrency=2)
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/tests/integration/test_graph_ops.py]: I extracted individual tests for these now

**Review by adamnsch — APPROVED** [review]
Very nice work :) Will be nice in when we can remove support for the old API

File: graphdatascience/graph/graph_entity_ops_runner.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        # old format was requested but the query was run via Arrow
+        elif not separate_property_columns and "propertyValue" not in result.keys():
+            result = result.melt(id_vars=["nodeId"]).rename(
+                columns={"variable": "nodeProperty", "value": "propertyValue"}
+            )
+
+        return result
+
+    @compatible_with("write", min_inclusive=ServerVersion(2, 2, 0))
+    def write(self, G: Graph, node_properties: List[str], node_labels: Strings = ["*"], **config: Any) -> Series:
```
Comment (FlorentinD) [inline graphdatascience/graph/graph_entity_ops_runner.py:64]: @soerenreichardt 
MyPy complains here `Returning Any from function declared to return "Series"`

Running `./scripts/checkstyle`.

Comment (FlorentinD) [reply graphdatascience/graph/graph_entity_ops_runner.py:64]: similar to previous call-sites we ignore the type warning here, as we know that write/drop will only return a single row

File: graphdatascience/graph/graph_entity_ops_runner.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        self._namespace += ".write"
+        return self._handle_properties(G, node_properties, node_labels, config).squeeze()  # type: ignore
+
+    @compatible_with("drop", min_inclusive=ServerVersion(2, 2, 0))
+    def drop(self, G: Graph, node_properties: List[str], **config: Any) -> DataFrame:
```
[the remark this answers is not in the document]
Comment (adamnsch) [reply graphdatascience/graph/graph_entity_ops_runner.py]: done

**Review by adamnsch — APPROVED** [review]

**Pull request merged**